heat/core/dndarray.py

```diff
@@ -867,6 +880,26 @@ def create_lshape_map(self):
         self.comm.Allreduce(MPI.IN_PLACE, lshape_map, MPI.SUM)
         return lshape_map
 
+    def create_square_diag_tiles(self, tiles_per_proc=None):
+        """
+        Create the tiles for the DNDarray as defined by the tiling.SquareDiagTiles class. These
+        tiles can be accessed as a property. For more imformation on these tiles see
+        :func:`SquareDiagTiles <tiling.SquareDaigTiles>`.
+
+        Parameters
+        ----------
+        tiles_per_proc : int, optional
+            Default = 2 (see :func:`SquareDiagTiles <tiling.SquareDaigTiles>`)
+            the number of divisions per process,
+
+        Returns
+        -------
+        None
+        """
+        self.__tiles = tiling.SquareDiagTiles(
+            self, tiles_per_proc=tiles_per_proc
+        )  # type: tiling.SquareDiagTiles
```

Comment:
Comment left over

Reply:
My IDE was have troubles inferring the type of tiles here so I have it explicitly defined it with a type tag to make things run more smoothly